file/file.go

```diff
@@ -0,0 +1,70 @@
+// Package file - Handle file operations
+package file
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+)
+
+const filename = "tasks.json"
+
+// File - file generic struct. T generic.
+type File[T any] struct{}
+
+// New - instantiates a File struct.
+func New[T any]() *File[T] {
+	return &File[T]{}
+}
+
+// Write - writes tasks into a JSON file
+func (f *File[T]) Write(data *[]T) error {
+	asJSON, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(filename, asJSON, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Read - reads tasks from json file
+func (f *File[T]) Read() ([]T, error) {
+	jsonFile, err := os.Open(filename)
+
+	if err != nil {
```

Comment:
no need for this if, errors.Is works fine with nil too. nil error is not os.ErrNotExists, so logic won't change if you remove it. so you should remove it.